test/all.test.js

```diff
@@ -618,6 +652,165 @@ describe(Support.getTestDialectTeaser('Tests'), function () {
+							//additional checks
+							expect(folder.drive).to.be.ok;
+							expect(folder.drive.name).to.equal(this.aDrive.name);
+							//scope is not applied to children unless we explicitly state it in the model definition
+							expect(folder.children[0].drive).to.be.falsy;
+						});
+				});
+
+				it('children with scope', function() {
+					return scopedModel.find({
+						where: {name: 'a'},
+						include: [{model: scopedModel, as: 'children'}],
+						order: [[{model: scopedModel, as: 'children'}, 'name']]
+					}).bind(this)
+						.then(function(folder) {
+							expect(folder.children.length).to.equal(2);
+							expect(folder.children[0].name).to.equal('ab');
+							expect(folder.children[1].name).to.equal('ac');
+							//additional checks
+							expect(folder.drive).to.be.ok;
+							expect(folder.drive.name).to.equal(this.aDrive.name);
+							//scope is not applied to children unless we explicitly state it in the model definition
+							expect(folder.children[0].drive).to.be.ok;
+							expect(folder.children[0].drive.name).to.equal(this.aDrive.name);
+						});
+				});
+
+				it('descendents', function() {
+					return scopedModel.find({
+						where: {name: 'a'},
+						include: [{model: this.folder, as: 'descendents'}],
+						order: [[{model: this.folder, as: 'descendents'}, 'name']]
+					}).bind(this)
+						.then(function(folder) {
+							expect(folder.descendents.length).to.equal(6);
+							expect(folder.descendents[0].name).to.equal('ab');
+							expect(folder.descendents[1].name).to.equal('abd');
+							expect(folder.descendents[2].name).to.equal('abdf');
+							expect(folder.descendents[3].name).to.equal('abdg');
+							expect(folder.descendents[4].name).to.equal('abe');
+							expect(folder.descendents[5].name).to.equal('ac');
+
+							expect(folder.drive).to.be.ok;
+							expect(folder.drive.name).to.equal(this.aDrive.name);
+						});
+				});
+
+				it('parent', function() {
+					return scopedModel.find({
+						where: {name: 'abdf'},
+						include: [{model: this.folder, as: 'parent'}]
+					}).bind(this)
+						.then(function(folder) {
+							expect(folder.parent).to.be.ok;
+							expect(folder.parent.name).to.equal('abd');
+
+							expect(folder.drive).to.be.ok;
+							expect(folder.drive.name).to.equal(this.aDrive.name);
+						});
+				});
+
+				it('ancestors', function() {
+					return scopedModel.find({
+						where: {name: 'abdf'},
+						include: [{model: this.folder, as: 'ancestors'}],
+						order: [[{model: this.folder, as: 'ancestors'}, 'hierarchyLevel']]
+					}).bind(this)
+						.then(function(folder) {
+							expect(folder.ancestors.length).to.equal(3);
+							expect(folder.ancestors[0].name).to.equal('a');
+							expect(folder.ancestors[1].name).to.equal('ab');
+							expect(folder.ancestors[2].name).to.equal('abd');
+
+							expect(folder.drive).to.be.ok;
+							expect(folder.drive.name).to.equal(this.aDrive.name);
+						});
+				});
+
```

Comment:
descendents in hierarchy & descendents with scope in hierarchy are the actual tests for #15